### PR TITLE
small refactor + adding intervals

### PR DESCRIPTION
**Pull request by maxulysse**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/references/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/references _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core pipelines lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Check for unexpected warnings in debug mode (`nextflow run . -profile debug,test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: assets/schema_input.json [hunk 1]
```diff
@@ -23,6 +23,11 @@
                 "errorMessage": "Species of the reference",
                 "meta": ["species"]
             },
+            "bed_intervals": {
+                "type": "string",
+                "pattern": "^\\S+\\.bed?$",
+                "errorMessage": "TODO"
```
Comment (LouisLeNezet) [inline assets/schema_input.json:29]: I think you need the `errorMessage` here.
Also can a bed file really be `.be` ?

File: tests/rnaseq.nf.test [hunk 2]
```diff
@@ -4,44 +4,13 @@ nextflow_pipeline {
     script "../main.nf"
     tag "pipeline"
 
-    test("Run with profile test for rnaseq | --input assets/genomes/test/pipelines/default_extended.yml") {
```
Comment (LouisLeNezet) [inline tests/rnaseq.nf.test:7]: Why did you delete this test ?

Comment (LouisLeNezet) [reply tests/rnaseq.nf.test:7]: Just saw that you move it to sarek test.

File: workflows/references/main.nf [hunk 3]
```diff
@@ -72,81 +79,76 @@ workflow REFERENCES {
...
-    }
-
-    if (tools && tools.split(',').contains('createsequencedictionary')) {
-        GATK4_CREATESEQUENCEDICTIONARY(input.fasta)
-
-        dict = GATK4_CREATESEQUENCEDICTIONARY.out.dict
-        versions = versions.mix(GATK4_CREATESEQUENCEDICTIONARY.out.versions)
-    }
+    // the whole map -> null should be removed once I managed to make it work properly
+    BWAMEM1_INDEX(
```
Comment (LouisLeNezet) [inline workflows/references/main.nf:83]: This is a really nice way to skip the repeated if statement.
On the other hand this will pollute the output nextflow log with empty processes...
Is it okay for you ?

File: workflows/references/main.nf [hunk 4]
```diff
@@ -72,81 +79,76 @@ workflow REFERENCES {
...
+            return file[1] ? null : [meta, file]
+        }
 
-        versions = versions.mix(SAMTOOLS_FAIDX.out.versions)
-    }
+    BUILD_INTERVALS(faidx_intervals, [])
 
-    if (tools && tools.split(',').contains('gffread')) {
+    if (tools.contains('gffread')) {
         GFFREAD(input.gff, [])
```
Comment (LouisLeNezet) [inline workflows/references/main.nf:145]: Why didn't you continue with the simplification of the if statements ?

File: assets/schema_input.json (hunk 1 above)
Comment (maxulysse) [reply assets/schema_input.json:29]: yeah, I need error messages everywhere, I'll do this later.
`.be`? it's `.bed` here, no?

File: workflows/references/main.nf (hunk 4 above)
Comment (maxulysse) [reply workflows/references/main.nf:145]: Just testing it out on some processes, I didn't want to modify everything in one go as I just wanted to tackle intervals on this PR.

File: workflows/references/main.nf (hunk 3 above)
Comment (maxulysse) [reply workflows/references/main.nf:83]: that's fine for now, I'll deal with this later if it's really an issue, but for now, I just want a code that is cleaner to read

File: assets/schema_input.json (hunk 1 above)
Comment (LouisLeNezet) [reply assets/schema_input.json:29]: If you use `.bed?` then the `d` is optional in the regex, normally.
You can try easily with https://regexr.com/

File: workflows/references/main.nf (hunk 3 above)
Comment (LouisLeNezet) [reply workflows/references/main.nf:83]: In phaseimpute we also have this issue with branching event, but nothing can be done.

File: workflows/references/main.nf (hunk 4 above)
Comment (LouisLeNezet) [reply workflows/references/main.nf:145]: That does make sense.
Could you open an issue in this case in the repository to have a follow up for standardization ?

File: assets/schema_input.json (hunk 1 above)
Comment (maxulysse) [reply assets/schema_input.json:29]: :facepalm: of course

File: assets/schema_input.json [hunk 5]
```diff
@@ -23,6 +23,11 @@
                 "errorMessage": "Species of the reference",
                 "meta": ["species"]
             },
+            "bed_intervals": {
+                "type": "string",
+                "pattern": "^\\S+\\.bed?$",
```
Comment (maxulysse) [inline assets/schema_input.json:28]:
```suggestion
                "pattern": "^\\S+\\.bed$",
```
